### PR TITLE
Document how a ooniprobe works at a high level

### LINKED ISSUE #138: Describe the probe operation at an high level
I'd like to have a short document explaining at a high level to devs how a probe works. This is motivated by the following rationale:

1. the desktop release may attract new devs, better to have an onboarding for devs doc

2. I'd like to write more specific docs regarding the engine, pulling them from MK because:

2.1. it will benefit me when working on the Go engine

2.2. it will benefit devs and integrators

This is tracked by MK's specific project for OONI desktop v3.0.0: https://github.com/orgs/measurement-kit/projects/2

I've already sketched out this document in a branch. @agrabeli seems very interested to help me moving this specific issue forward, thanks!

### PR DESCRIPTION
**Pull request by bassosimone**
Closes #138.

The idea is to provide a bird eyer's view to new developers.

### CODE REVIEW

File: probe/README.md [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+
+The [collector](../backends/bk-003-collector.md)
+is a set of servers to which the probe submits the
+results of nettests.
+
+### Test helpers
+
+The test helpers are a set of servers useful to perform specific nettests. Their
+specs is available [as part of this repository](../backends). We only consider
+test helpers the servers that are under OONI control. Other servers will be
```
Comment (sarathms) [inline probe/README.md]: This statement sounds a bit confusing. One reference is to servers under OONI's control, and the other set of servers which we use for testing, implying OONI controls them as well.

File: probe/README.md [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+
+### Geolookup
+
+Unless configured to skip this step, the engine will perform a geolookup
+as shown in interaction (2). The purpose of geolookup is to know the
+user IP, which is by default not included into the report, and information
+that can be guessed from the IP, like the ASN, the CC, etc. Knowing the
+IP also allows the engine to attempt to scrub the IP from the results, when
+the user has requested the engine not to include their IP (which is the
+default). In this document we don't want to the details of our [data
```
Comment (sarathms) [inline probe/README.md]:
```suggestion
default). In this document we don't want to get into the details of our [data
```

File: probe/README.md [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+..//nettests/ts-017-web-connectivity.md), this will require one or more URLs
+as input. The engine will access those URLs and use a specific test helpers
+to also access those URLs and have a comparison. The results of comparing
+the engine and the test helper measurement will become the result of the
+web measurement;
+
+2. if you run a [NDT test](../nettests/ts-022-ndt.md), there will be no input
+and no OONI controlled test helper. However, the test will measure the performance
+between the engine and a measurement server (which we don't consider a
+test helper because is not directly controlled by OONI). The performance
```
Comment (sarathms) [inline probe/README.md]: Would it be useful to mention or link to who owns the measurement servers?

File: probe/README.md (hunk 1 above)
Comment (bassosimone) [reply probe/README.md]: Addressed

File: probe/README.md (hunk 2 above)
Comment (bassosimone) [reply probe/README.md]: Already addressed by @agrabeli

File: probe/README.md (hunk 3 above)
Comment (bassosimone) [reply probe/README.md]: Addressed

**Pull request merged**